### PR TITLE
Fix jdbc date conversion offset 1 day

### LINKED ISSUE #21115: JdbcIO date conversion is sensitive to OS
Time handling varies between OSes:

for e.g.
```

// DB Init script
CREATE TABLE SimpleFlatRecords
(
    id                    INTEGER AUTO_INCREMENT
PRIMARY KEY,
    birth_date            DATE
);
INSERT INTO SimpleFlatRecords(id, birth_date)
VALUES
(1, '2000-05-11'),
       (2, '2001-06-12'),
       (3, '2002-07-13');
```

The Beam Row contains different dates based on OS:

MacOS:

1: '2000-05-10'
2: '2001-06-11'
3: '2002-07-12'

Linux:

1: '2000-05-11'
2: '2001-06-12'
3: '2002-07-13'

Imported from Jira [BEAM-12484](https://issues.apache.org/jira/browse/BEAM-12484). Original Jira may contain additional context.
Reported by: anant.damle.

### PR DESCRIPTION
**Pull request by Abacn**
Fixes #21115

* Use java.time.LocalDate directly instead of obtaining it from
  legacy java.sql.Date

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Abacn** [comment]
Tested on MacOS 12.5/M1

setup table:
```
postgres=# create table datetest(f_id INTEGER, f_date DATE);
CREATE TABLE
postgres=# insert into datetest values(1, '2022-08-16'),(2, NULL);
INSERT 0 2
postgres=# select * from datetest;
 f_id |   f_date   
------+------------
    1 | 2022-08-16
    2 | 
(2 rows)
```

v2.40.0:
```
Row: 
f_id:2
f_time:<null>

Row: 
f_id:1
f_time:2022-08-15T00:00:00.000Z
```

With change:
```
Row: 
f_id:2
f_date:<null>

Row: 
f_id:1
f_date:2022-08-16T00:00:00.000Z
```

**Comment by Abacn** [comment]
R: @johnjcasey

File: sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/SchemaUtil.java [hunk 1]
```diff
@@ -313,11 +311,11 @@ private static <InputT, BaseT> ResultSetFieldExtractor createLogicalTypeExtracto
   /** Convert SQL date type to Beam DateTime. */
   private static ResultSetFieldExtractor createDateExtractor() {
     return (rs, i) -> {
-      Date date = rs.getDate(i, Calendar.getInstance(TimeZone.getTimeZone(ZoneOffset.UTC)));
+      java.time.LocalDate date = rs.getObject(i, java.time.LocalDate.class);
```
Comment (johnjcasey) [inline sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/SchemaUtil.java:314]: can we change this to an import?

Comment (Abacn) [reply sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/SchemaUtil.java:314]: Unfortunately there is import of `org.joda.time.LocalDate` and used everywhere. There is an issue #19215 to eliminate joda.time in release 3.0.0

Comment (johnjcasey) [reply sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/SchemaUtil.java:314]: Gross, but understandable. Can you add a TODO here linked to that issue to clean this up?

**Comment by johnjcasey** [comment]
LGTM, I'll merge once the java test passes

**Comment by Abacn** [comment]
@johnjcasey Tests passed

**Pull request merged**